storage_s3/indico_storage_s3/plugin.py

```diff
@@ -0,0 +1,133 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2018 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+import sys
+from contextlib import contextmanager
+from datetime import datetime
+from tempfile import NamedTemporaryFile
+
+import boto3
+from werkzeug.datastructures import Headers
+from werkzeug.utils import redirect
+
+from indico.core import signals
+from indico.core.config import config
+from indico.core.plugins import IndicoPlugin
+from indico.core.storage import Storage, StorageError
+
+
+class S3StoragePlugin(IndicoPlugin):
+    """S3 Storage
+
+    Provides S3 storage backends.
+    """
+
+    def init(self):
+        super(S3StoragePlugin, self).init()
+        self.connect(signals.get_storage_backends, self._get_storage_backends)
+
+    def _get_storage_backends(self, sender, **kwargs):
+        return S3Storage
+
+
+class S3Storage(Storage):
+    name = 's3'
+    simple_data = False
+
+    def __init__(self, data):
+        data = self._parse_data(data)
+        self.host = data['host']
+        self.bucket = self.get_bucket_name(data['bucket'])
+        self.secret_key = data['secret_key']
+        self.access_key = data['access_key']
+        if any(x in data['bucket'] for x in ['<year>', '<month>', '<week>']):
+            self.acl_template_bucket = data['acl_template_bucket']
```

Comment:
shouldn't this be optional? maybe not everyone wants to set a custom ACL on their bucket.